### PR TITLE
add heating element

### PR DESCRIPTION
**Pull request by JasonMWhite**
This PR adds the long-awaited Heating element. It isn't yet included in the actual transform operation, merely the ability to parse it from xml and serialize it with `to_dict`.

There are some known holes that I was not able to yet figure out - no known examples of electric baseboards, P9-11 efficiency heaters, or combo heating/water heater systems. The parsing logic should fail loudly with explicit error messages in these cases.

### CODE REVIEW

File: etl/src/energuide/embedded/heating.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        return capacity
+
+    @classmethod
+    def _get_heating_type(cls, node: element.Element) -> HeatingType:
+        candidates = [candidate.tag for candidate in node.xpath('Type1/*')]
+        heating_type: typing.Optional[HeatingType] = None
+        for candidate in candidates:
+            if candidate in cls._HEATING_TYPE_NODE_NAMES:
+                heating_type = cls._HEATING_TYPE_NODE_NAMES[candidate]
+                break
```
Comment (buckley-w-david) [inline etl/src/energuide/embedded/heating.py]: This section seems odd to me, I'm guessing there's a good reason for it, possibly motivated by lack of test data?

File: etl/tests/embedded/test_heating.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            </Mode>
+            <Power isCalculated="true" low="0" high="145.5" />
+        </FansAndPump>
+    </Type1>
+    <Type2 shadingInF280Cooling="AccountedFor" />
+</HeatingCooling>
+    """
+    node = element.Element.from_string(data)
+    type1_node: element.Element = node.xpath('Type1')[0]
+    heating_node = sample_heating_node(heating_type)
```
Comment (buckley-w-david) [inline etl/tests/embedded/test_heating.py]: This seems odd to have down here when the other node creation calls are done at the top of the function

File: etl/src/energuide/embedded/heating.py (hunk 1 above)
Comment (JasonMWhite) [reply etl/src/energuide/embedded/heating.py]: Yes, we don't know what all the heating type node names are. There are multiple possible child nodes of `Type1`, only one of them is the one we want, and we don't know all the names. The three we know are `Baseboards`, `Furnace`, and `Boiler`. I haven't been able to figure out the other 2 yet, so I intentionally want it to fail and raise an error.

File: etl/tests/embedded/test_heating.py (hunk 2 above)
Comment (JasonMWhite) [reply etl/tests/embedded/test_heating.py]: At the top, if they are `None` they get set to their default value. Should I move this line up there with them for consistency, although it's slightly different?

Comment (buckley-w-david) [reply etl/tests/embedded/test_heating.py]: I'll leave it up to you, if I were writing it I'd put it at the top with the others, but that seems more like an personal preference thing

**Pull request merged**